packages/ui/src/common/providers/onboarding/provider.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react'
+
+import { useMyAccounts } from '@/accounts/hooks/useMyAccounts'
+import { useApi } from '@/common/hooks/useApi'
+import { useLocalStorage } from '@/common/hooks/useLocalStorage'
+import { useToggle } from '@/common/hooks/useToggle'
+import { OnBoardingContext } from '@/common/providers/onboarding/context'
+import { UseMembershipOnBoarding } from '@/common/providers/onboarding/types'
+import { useMyMemberships } from '@/memberships/hooks/useMyMemberships'
+
+interface Props {
+  children: React.ReactNode
+}
+
+export const OnBoardingProvider = (props: Props) => {
+  const [membershipModal, toggleMembershipModal] = useToggle()
+  const [membershipOnBoardingValue, setMembershipOnBoardingValue] = useLocalStorage<string>('onboarding-intro')
+  const membership = useMembershipOnBoarding(membershipOnBoardingValue)
+
+  const value = {
+    membership: {
+      ...membership,
+      setFreeTokens: setMembershipOnBoardingValue,
+      isModalOpen: membershipModal,
+      toggleModal: toggleMembershipModal,
```

Comment:
I think that those two are not related to the onboarding status. So I'd see them outside this hook.

Looking at how the components are used together _it should be_ possible to have the `useToggle()` hook defined in `<MembershipOnBoarding />` or in `<OnBoardingOverlay />`.

Reply:
Transferred to parent component of modal and overlay